pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -209,6 +227,65 @@ func (cd *clusterDeploy) deployAgent(cluster *v3.Cluster) error {
 	return err
 }
 
+func (cd *clusterDeploy) validateAgent(cluster *v3.Cluster) error {
```

Comment:
The flow of this should be updated:
Get the cluster with the read lock
Do all validation on the cluster
Assuming no errors, grab a write lock and update the local cache
Currently this grabbing and dropping locks then would grab the lock again to delete from the cache.

Reply:
This should also be integrated into the `redeployAgent` func.